plugin-core/src/main/java/appland/telemetry/appinsights/AppInsightsTelemetryReporter.java

```diff
@@ -24,13 +27,18 @@ public class AppInsightsTelemetryReporter implements TelemetryReporter {
     private static final @NotNull String DEFAULT_INGESTION_ENDPOINT = "https://centralus-0.in.applicationinsights.azure.com";
 
     private final @NotNull String ingestionEndpoint;
+    private final @NotNull Map<String, String> commonProperties;
 
-    public AppInsightsTelemetryReporter() {
-        this(DEFAULT_INGESTION_ENDPOINT);
+    public AppInsightsTelemetryReporter(@NotNull Map<String, String> commonProperties) {
+        this(DEFAULT_INGESTION_ENDPOINT, commonProperties);
     }
 
-    AppInsightsTelemetryReporter(@NotNull String ingestionEndpoint) {
+    AppInsightsTelemetryReporter(@NotNull String ingestionEndpoint, @NotNull Map<String, String> commonProperties) {
         this.ingestionEndpoint = ingestionEndpoint;
+
+        var prefixedProperties = new HashMap<String, String>();
+        commonProperties.forEach((key, value) -> prefixedProperties.put("common." + key, value));
+        this.commonProperties = prefixedProperties;
```

Comment:
Perhaps a bit nicer and creating an unmodifiable map:
```suggestion
        this.commonProperties = commonProperties.entrySet().stream()
                .collect(Collectors.toUnmodifiableMap(key -> "common." + key, Map.Entry::getValue));
```